src/Transports.AspNetCore/ExecutionResultActionResult.cs

```diff
@@ -0,0 +1,34 @@
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// An action result that formats the <see cref="ExecutionResult"/> as JSON.
+/// </summary>
+public sealed class ExecutionResultActionResult : IActionResult
+{
+    private readonly ExecutionResult _executionResult;
+    private readonly HttpStatusCode _statusCode;
+
+    /// <inheritdoc cref="ExecutionResultActionResult"/>
+    public ExecutionResultActionResult(ExecutionResult executionResult)
+    {
+        _executionResult = executionResult;
+        _statusCode = executionResult.Executed ? HttpStatusCode.OK : HttpStatusCode.BadRequest;
+    }
+
+    /// <inheritdoc cref="ExecutionResultActionResult"/>
+    public ExecutionResultActionResult(ExecutionResult executionResult, HttpStatusCode statusCode)
+    {
+        _executionResult = executionResult;
+        _statusCode = statusCode;
+    }
+
+    /// <inheritdoc/>
+    public async Task ExecuteResultAsync(ActionContext context)
+    {
+        var writer = context.HttpContext.RequestServices.GetRequiredService<IGraphQLSerializer>();
+        var response = context.HttpContext.Response;
+        response.ContentType = GraphQLHttpMiddleware.CONTENTTYPE_GRAPHQLJSON;
```

Comment:
Since `SelectResponseContentType` is virtual I suggest to add get/set public property here with default value of `GraphQLHttpMiddleware.CONTENTTYPE_GRAPHQLJSON`.

Reply:
At least it is consistent with middleware code.